vega_sim/scenario/common/agents.py

```diff
@@ -889,24 +897,30 @@ def step(self, vega_state: VegaState):
                 Object describing the state of the network and the market.
         """
 
-        if (
-            vega_state.market_state[self.market_id].trading_mode
-            == markets_protos.Market.TradingMode.TRADING_MODE_CONTINUOUS
-        ) and vega_state.market_state[
-            self.market_id
-        ].state == markets_protos.Market.State.STATE_ACTIVE:
+        self.current_step += 1
 
-            if self.random_state.rand() <= self.submit_bias:
-                self._submit_order()
+        if self.random_state.rand() <= self.submit_bias:
+            self._submit_order(vega_state=vega_state)
 
-            if self.random_state.rand() <= self.cancel_bias:
-                self._cancel_order(vega_state=vega_state)
+        if self.random_state.rand() <= self.cancel_bias:
+            self._cancel_order(vega_state=vega_state)
 
-    def _submit_order(self):
+    def _submit_order(self, vega_state: VegaState):
 
-        best_bid_price, best_offer_price = self.vega.best_prices(
-            market_id=self.market_id
-        )
+        # Calculate mean_buy_price and mean_sell_price of price distribution
+        if (isinstance(self.spread, type(None))) or (
+            isinstance(self.price_process, type(None))
+        ):
+            # If agent does not have price_process data, offset orders from best bid/ask
+            best_bid_price, best_offer_price = self.vega.best_prices(
+                market_id=self.market_id
+            )
+            mean_buy_price = best_bid_price
+            mean_sell_price = best_offer_price
+        else:
+            # If agent does have price_process data, offset orders from market price
+            mean_buy_price = self.price_process[self.current_step] - self.spread / 2
+            mean_sell_price = self.price_process[self.current_step] + self.spread / 2
```

Comment:
I like the logic, but can we rename from `mean` since it isn't really a mean? `buy_price_reference` or something?